envi/symstore/resolver.py

```diff
@@ -201,15 +202,20 @@ def delSymbol(self, sym):
             subres = self.symnames.get(sym.fname)
 
         # Potentially del it from the sub resolver's namespace
-        if subres is not None and not isinstance(subres, tuple):
+        if subres and isinstance(subres, es_symcache.SymbolCache):
             subres.delSymbol(sym)
 
         # Otherwise del it from our namespace
         else:
             symname = sym.name
             if not self.casesens:
                 symname = symname.lower()
-            self.symnames.pop(symname, None)
+
```

Comment:
these are important bugfixes...

Reply:
POSIX and Windows use different methods, this allows the correct deletion from both.